forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/utils/AccountDataInternalIdFilter.java

```diff
@@ -25,124 +25,278 @@
+ * not displayed), or leaves it populated.
+ *
+ * <p>
+ * By default, the config property <code>rs.data.internal_ids.show</code> is set to <code>true</code>, meaning all IDs
+ * remain populated. If some of the internal IDs should be hidden, then <code>rs.data.internal_ids.show</code> should
+ * be set to <code>false</code> and the preference for each relevant object should be specified (e.g.
+ * <code>rs.data.internal_ids.direct_debit.show</code> for a Direct Debit).
+ */
 @Service
 @Slf4j
 public class AccountDataInternalIdFilter {
-    private final boolean showAccountDataInternalIds;
 
-    public AccountDataInternalIdFilter(@Value("${rs.data.internal_ids.show:true}") boolean showAccountDataInternalIds) {
-        this.showAccountDataInternalIds = showAccountDataInternalIds;
+    private final boolean showAllAccountDataInternalIds;
+    private final boolean showDirectDebitIds;
+    private final boolean showOfferIds;
+    private final boolean showProductIds;
+    private final boolean showStatementIds;
+    private final boolean showDomesticTransactionIds;
+    private final boolean showInternationalTransactionIds;
+    private final boolean showDomesticBeneficiaryIds;
+    private final boolean showInternationalBeneficiaryIds;
+    private final boolean showDomesticScheduledPaymentIds;
+    private final boolean showInternationalScheduledPaymentIds;
+    private final boolean showDomesticStandingOrderIds;
+    private final boolean showInternationalStandingOrderIds;
+
+    public AccountDataInternalIdFilter(
+            @Value("${rs.data.internal_ids.show:true}") boolean showAllAccountDataInternalIds,
+            @Value("${rs.data.internal_ids.direct_debit.show:false}") boolean showDirectDebitIds,
+            @Value("${rs.data.internal_ids.offer.show:false}") boolean showOfferIds,
+            @Value("${rs.data.internal_ids.product.show:false}") boolean showProductIds,
+            @Value("${rs.data.internal_ids.statement.show:true}") boolean showStatementIds,
+            @Value("${rs.data.internal_ids.domestic.transaction.show:true}") boolean showDomesticTransactionIds,
+            @Value("${rs.data.internal_ids.international.transaction.show:true}") boolean showInternationalTransactionIds,
+            @Value("${rs.data.internal_ids.domestic.beneficiary.show:false}") boolean showDomesticBeneficiaryIds,
+            @Value("${rs.data.internal_ids.international.beneficiary.show:true}") boolean showInternationalBeneficiaryIds,
+            @Value("${rs.data.internal_ids.domestic.scheduled_payment.show:false}") boolean showDomesticScheduledPaymentIds,
+            @Value("${rs.data.internal_ids.international.scheduled_payment.show:false}") boolean showInternationalScheduledPaymentIds,
+            @Value("${rs.data.internal_ids.domestic.standing_order.show:false}") boolean showDomesticStandingOrderIds,
+            @Value("${rs.data.internal_ids.international.standing_order.show:false}") boolean showInternationalStandingOrderIds) {
```

Comment:
These default values have been requested by one of our customers, but I'm going to raise a PR to set them explicitly in their config.

I'm in two minds about removing these defaults (except for the `rs.data.internal_ids.show` which pre-dates this PR and was already defaulted to true)

Reply:
> Looks good Matt - could you add the default settings to the openbanking-reference-implementation spring config with a comment to say what they control an their effects? Ta!

Covered by this PR: https://github.com/ForgeCloud/ob-sandbox-config/pull/20

Reply:
I've set them all to false because I'm not keen on the idea of customer specific defaults in the code. However, the defaults are set according to LBG in the rs-store config.